### ISSUE TITLE
Module request: Flexbar

### ISSUE THREAD

**Issue by astewart-twist** [ewels/MultiQC#472]
_- Description of tool, link to homepage_

https://github.com/seqan/flexbar

The program Flexbar preprocesses high-throughput sequencing data efficiently. It demultiplexes barcoded runs and removes adapter sequences. Moreover, trimming and filtering features are provided. Flexbar increases read mapping rates and improves genome as well as transcriptome assemblies. It supports next-generation sequencing data in fasta and fastq format, e.g. from Roche 454 and the Illumina platform.

_- Example log files?_
https://github.com/ewels/MultiQC_TestData/pull/35

_- Any standard log file naming patterns to get sample name?_
Optionally specified at command line.

_- Which value is most important, to go into the General Stats table?_
`Remaining reads` ?

Could probably follow the same pattern as Trimmomatic. 

_- What should be plotted, if anything?_
Again, probably the same as Trimmomatic.  They're very similar tools.

**Comment by ewels** [comment]
Hi @astewart-twist,

I've just had a stab at writing this module for you. If you could grab a copy of the latest development version of MultiQC and see what you think that would be great!

Cheers,

Phil

**Comment by astewart-twist** [comment]
Thanks @ewels !  Will do!

**Comment by astewart-twist** [comment]
Hey @ewels , looks like multiqc can't detect the flexbar file?

```
multiqc .
[INFO   ]         multiqc : This is MultiQC v1.1.dev0
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching '.'
Searching 1 files..  [####################################]  100%
[WARNING]         multiqc : No analysis results found. Cleaning up..
[INFO   ]         multiqc : MultiQC complete
```

How is it scanning for it?

**Comment by ewels** [comment]
Hi @astewart-twist,

It looks for any file with the string `'Flexbar - flexible barcode and adapter removal'` (see the [search patterns config](https://github.com/ewels/MultiQC/blob/master/multiqc/utils/search_patterns.yaml#L61-L63)).

How did you update MultiQC for testing? Note that if you have cloned MultiQC and pulled updates, then you need to rerun `python setup.py` for the module to run. If you used `pip` and already had `v1.1.dev0` then it won't update by default (as it already has the latest version number) and you need to use the `--upgrade --force-reinstall` flags (see the readme for the full command).

Phil

**Comment by astewart-twist** [comment]
I'm using `pip install --upgrade --force-reinstall git+https://github.com/ewels/MultiQC.git` to install the update.

**Comment by ewels** [comment]
This is odd. Are the files you're parsing different to the one that you submitted to the TestData repository? That one is working fine for me:

```
$ multiqc -f data/ -m flexbar

[INFO   ]         multiqc : This is MultiQC v1.1.dev0 (45642d6)
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching 'data/'
[INFO   ]         multiqc : Only using modules flexbar
Searching 1175 files..  [####################################]  100%
[INFO   ]         flexbar : Found 1 logs
[INFO   ]         multiqc : Compressing plot data
[WARNING]         multiqc : Deleting    : multiqc_report.html   (-f was specified)
[WARNING]         multiqc : Deleting    : multiqc_data   (-f was specified)
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data
[INFO   ]         multiqc : MultiQC complete
```
(you can see it running in the [automated tests](https://travis-ci.org/ewels/MultiQC/jobs/252018407#L633) too)..

Could you try running with `-m flexbar` to confirm that you definitely have the flexbar module in your copy of MultiQC? If that works ok, could you run in verbose mode (`-v`) and send me the log, and any files that you're trying to work with please?

Cheers,

Phil

**Comment by astewart-twist** [comment]
Hey @ewels :  `-m flexbar` and `-v` confirm that I have the flexbar module in my current dev version (that was my first suspicion too).  

I think the issue might be some slight differences in the format from a more recent version of flexbar?  I noticed the file I'm looking at has:

```
Flexbar - flexible barcode and adapter removal, version 2.5
```

while other log files I have from before contain

```
Flexbar - flexible barcode and adapter removal
```


I'll sanitize the file I'm currently having problems on and share it with you.

**Comment by astewart-twist** [comment]
```

               ________          __
              / ____/ /__  _  __/ /_  ____ ______
             / /_  / / _ \| |/ / __ \/ __ `/ ___/
            / __/ / /  __/>  </ /_/ / /_/ / /
           /_/   /_/\___/_/|_/_.___/\__,_/_/

Flexbar - flexible barcode and adapter removal


Local time:            Fri Jul  7 03:07:34 2017

Target name:           flexbar_out_dir/Sample1
File type:             fasta
Reads file:            /sbgenomics/Projects/xxx/Sample1.fasta

threads:               2
max-uncalled:          0
pre-trim-left:         22
pre-trim-right:        18
post-trim-length:      50
min-read-length:       18


Processing reads ...done.

Computation time:  < 1 sec


Output file statistics
======================
Read file:               flexbar_out_dir/Sample1.fasta
  written reads          46464
  skipped short reads    0


Filtering statistics
====================
Processed reads                   46464
  skipped due to uncalled bases       0
  finally skipped short reads         0
Discarded reads overall               0
Remaining reads                   46464   (100% of input)

Processed bases:   4181730
Remaining bases:   2323170   (55% of input)


Flexbar completed basic processing.

```

**Comment by ewels** [comment]
Thanks @astewart-twist - it should work now. The problem was due to differences in the log file format as you say. Specifically, the module was looking for the line `Flexbar completed adapter removal` to know where the end of the log file (useful if collecting from a `stdout` file). In the log file you pasted above this is instead `Flexbar completed basic processing.`. This meant that the module never saved the parsed data. This is fixed now - it just looks for `Flexbar completed` and also saves any data when it reaches the end of the log file.

There were also a couple of other minor things, such as an additional `:` character after `Processed bases`, but I've cleared them up too so it should all work.

Let me know how you get on with the updated version!

Thanks,

Phil
